docs/ota-updates.md

```diff
@@ -0,0 +1,51 @@
+# Over The Air Unattended Updates (OTAU2)
+
+This document enumerates some of the approaches and FLOSS software that could
+be used to deploy OTAU2 to Lepidopter distribution.
+
+## Requirements
+
+A list of Lepidopter's OTAU2 requirements:
+
+* Atomic software release update
+
+* On failure, deploy previous working bootloader, kernel
+    configuration, and filesystems
+
+* On success, deploy newest working bootloader, kernel
+    configuration, filesystems and reboot (if needed) for the changes to take
+    effect
+
+* Update of bootloader, kernel and configuration data, and filesystems
+
+* Support for signing of images and verification of images on
+    installation
+
+* Support for a self-hosted deployment server
+
+* Enable/disable a specific feature and apply/rollback system updates
+    incrementally rather than through a complete OS update that
+    replaces the filesystem
```

Comment:
There is another gotcha: your OS image may grow larger then temporary storage to download it. That's the reason to stream it from the network straight to the standby-root partition.